src/main/java/com/google/jenkins/plugins/storage/ClassicUpload.java

```diff
@@ -34,173 +33,185 @@
- * to the storage bucket.
+ * This upload extension implements the classical upload pattern where a user
+ * provides an Ant-style glob, e.g. ** / *.java relative to the build workspace,
+ * and those files are uploaded to the storage bucket.
  */
 public class ClassicUpload extends AbstractUpload {
 
-  /**
-   * Construct the classic upload implementation from the base properties
-   * and the glob for matching files.
-   */
-  @DataBoundConstructor
-  public ClassicUpload(String bucket, @Nullable UploadModule module,
-      String pattern,
-      // Legacy arguments for backwards compatibility
-      @Deprecated @Nullable String bucketNameWithVars,
-      @Deprecated @Nullable String sourceGlobWithVars) {
-    super(Objects.firstNonNull(bucket, bucketNameWithVars), module);
-    this.sourceGlobWithVars =
-        Objects.firstNonNull(pattern, sourceGlobWithVars);
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public String getDetails() {
-    return getPattern();
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  @Nullable
-  protected UploadSpec getInclusions(Run<?, ?> run,
-      FilePath workspace, TaskListener listener) throws UploadException {
-    try {
-
-      String globResolvedVars = StorageUtil
-          .replaceMacro(getPattern(), run, listener);
-
-      // In order to support absolute globs (e.g. /gagent/metaOutput/std*.txt)
-      // we must identify absolute paths and rebase the "workspace" to be the
-      // root directory and the glob to be relative to root.
-      //
-      // NOTE: FilePath#list complains when the ant-style glob is actually
-      // an absolute path.
-      //
-      // TODO(mattmoor): This will not work on Windows, but short of relying
-      // on Java 7's java.nio.file.Paths there doesn't appear to be a good
-      // platform agnostic approach to this.
-      if (globResolvedVars.startsWith("/")) {
-        workspace = getRoot(workspace);
-        // TODO(mattmoor): Support Windows
-        // Drop the leading "/" in a fashion that should be compatible with
-        // non-Unix platforms (e.g. Windows, which uses drive letters: C:\)
-        globResolvedVars = globResolvedVars.substring(1);
-      }
-
-      FilePath[] inclusions = workspace.list(globResolvedVars);
-      if (inclusions.length == 0) {
-        listener.error(module.prefix(
-            Messages.ClassicUpload_NoArtifacts(
-                globResolvedVars)));
-        return null;
-      }
-      listener.getLogger().println(module.prefix(
-          Messages.ClassicUpload_FoundForPattern(
-              inclusions.length, getPattern())));
-      return new UploadSpec(workspace, Arrays.asList(inclusions));
-    } catch (InterruptedException e) {
-      throw new UploadException(Messages.AbstractUpload_IncludeException(), e);
-    } catch (IOException e) {
-      throw new UploadException(Messages.AbstractUpload_IncludeException(), e);
+    public static <T> T firstNonNull(T... objs) {
```

Comment:
Was there an issue migrating to MoreObjects.firstNonNull(T, T) ?

Reply:
Because it's not existing in Java 1.8, not supported and removed.
It's not predictable function. Replaced it's with stupid fix, adding firstNonNull function.

Reply:
Objects.firstNonNull is deprecated and removed in the new version. The recommended migration path is MoreObjects.firstNonNull, did you try that?

Reply:
MoreObjects.firstNonNull - doesn't exists.